orc8r/cloud/go/syncstore/store_writer.go

```diff
@@ -0,0 +1,393 @@
+		invalidByNetwork[network] = keys
+	}
+
+	for _, network := range tracked {
+		filter := blobstore.CreateSearchFilter(&network, []string{cacheWriterBlobstoreType}, nil, nil)
+		criteria := blobstore.LoadCriteria{LoadValue: true}
+		networkBlobs, err := store.Search(filter, criteria)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "list outdated cache writers of existing network %+v", network))
+			continue
+		}
+
+		invalid := []string{}
+		for _, blob := range networkBlobs[network] {
+			creationTime := binary.LittleEndian.Uint64(blob.Value)
+			if clock.Now().Unix()-int64(creationTime) > cacheWriterValidIntervalSecs {
+				invalid = append(invalid, blob.Key)
+			}
+		}
+		invalidByNetwork[network] = invalid
+	}
+	return invalidByNetwork, errs.ErrorOrNil()
+}
+
+// dropInvalidCaches drops the temporary caches held by invalid cache writers, and
+// returns the IDs of those successfully dropped.
+func dropInvalidCaches(db *sql.DB, invalidByNetwork map[string][]string) (map[string][]string, error) {
+	errs := &multierror.Error{}
+	deletedByNetwork := map[string][]string{}
+	for network, invalid := range invalidByNetwork {
+		deleted := []string{}
+		for _, tableName := range invalid {
+			txFn := func(tx *sql.Tx) (interface{}, error) {
+				_, err := tx.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %s", tableName))
+				return nil, errors.Wrapf(err, "drop cache writer table %+v for network %+v", tableName, network)
+			}
+			_, err := sqorc.ExecInTx(db, nil, nil, txFn)
+			if err != nil {
+				multierror.Append(errs, err)
+				continue
+			}
+			deleted = append(deleted, tableName)
+		}
+		deletedByNetwork[network] = deleted
+	}
+	return deletedByNetwork, errs.ErrorOrNil()
+}
+
+// deleteCacheWriterBlobstoreRecords removes the blobstore records of cache writers
+// that are invalid, and whose temporary caches have been dropped.
+func deleteCacheWriterBlobstoreRecords(fact blobstore.BlobStorageFactory, deletedByNetwork map[string][]string) error {
+	store, err := fact.StartTransaction(nil)
+	if err != nil {
+		return errors.Wrap(err, "error starting transaction")
+	}
+	defer store.Rollback()
+
+	errs := &multierror.Error{}
+	for network, deleted := range deletedByNetwork {
+		tks := storage.MakeTKs(cacheWriterBlobstoreType, deleted)
+		err := store.Delete(network, tks)
+		if err != nil {
+			multierror.Append(errs, errors.Wrapf(err, "delete blobstore cache writer records %+v for network %+v", deleted, network))
+		}
+	}
+	err = store.Commit()
+	if err != nil {
+		multierror.Append(errs, err)
+	}
+
+	return errs.ErrorOrNil()
+}
+
+// generateCacheWriterUUID returns a universally unique ID for a cache writer.
+func generateCacheWriterUUID(tableNamePrefix string) string {
+	// Replace "-" symbols in the ID since they aren't supported by SQL variable names
+	id := strings.Replace((&storage.UUIDGenerator{}).New(), "-", "_", -1)
+	return fmt.Sprintf("%s_cache_writer_%s", tableNamePrefix, id)
+}
+
+func encodeUint64ToBytes(n uint64) []byte {
```

Comment:
Nit: can just call this encodeUint64, it's mostly implicit that you'd be encoding it to bytes